StoreKitUnitTests/TrialOrIntroPriceEligibilityCheckerSK2Tests.swift

```diff
@@ -131,4 +131,122 @@ class TrialOrIntroPriceEligibilityCheckerSK2Tests: StoreKitConfigTestCase {
             expect(receivedEligibility.status) == expected[product]
         }
     }
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    func testCheckEligibilityForProductIsEligibleForEligibleSubscription() async throws {
+        try AvailabilityChecks.iOS15APIAvailableOrSkipTest()
+
+        let productIdentifiers = Set([
+            "com.revenuecat.monthly_4.99.1_week_intro"
+        ])
+
+        let sk2Product = try await ProductsFetcherSK2().products(identifiers: productIdentifiers).first
+        let receivedProduct = try XCTUnwrap(sk2Product)
+        let storeProduct = StoreProduct.from(product: receivedProduct)
+
+        var completionCalled = false
+
+        let status: IntroEligibilityStatus = try await withCheckedThrowingContinuation({ continuation in
+            self.trialOrIntroPriceEligibilityChecker.checkEligibility(product: storeProduct) { status in
+                completionCalled = true
+                continuation.resume(returning: status)
+            }
+        })
+
+        expect(completionCalled) == true
+        expect(status) == .eligible
+    }
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    func testCheckEligibilityForLifetimeProductIsNoIntroOfferExists() async throws {
+        try AvailabilityChecks.iOS15APIAvailableOrSkipTest()
+
+        let productIdentifiers = Set([
+            "lifetime"
+        ])
+
+        let sk2Product = try await ProductsFetcherSK2().products(identifiers: productIdentifiers).first
+        let receivedProduct = try XCTUnwrap(sk2Product)
+        let storeProduct = StoreProduct.from(product: receivedProduct)
```

Comment:
We could make `StoreKitConfigTestCase.fetchSk2StoreProduct` take a product identifier that defaults to `Self.productID` to simplify this here too. I'm happy to send a follow up PR with that though.

Reply:
https://revenuecats.atlassian.net/browse/CF-288